### ISSUE TITLE
[Android] requestRuntimePermission and resume. Two executions.

### ISSUE THREAD

**Issue by gurumelo** [dpa99c/cordova-diagnostic-plugin#91]
Hello. requestRuntimePermission. If you "Allow" permission. Two executions. callback and resume event. How can i control this and only one execution? Thanks

CODE
```
document.addEventListener("deviceready", vamos, false);
document.addEventListener("resume", permisos, false);

function vamos() {
	iniciarbase();
	permisos();
}

var estado = 0;
var posicionboton = 0;

function permisos(){
    console.log('ejecuto PERMISOS!!!');
    cordova.plugins.diagnostic.requestRuntimePermission(function(status){
        switch(status){
            case cordova.plugins.diagnostic.runtimePermissionStatus.GRANTED:
                console.log("Permission granted (or already granted) - call the plugin");
                relocalizar();
                console.log('Estoy dentro de GRANTED!!!!!');
                break;
            case cordova.plugins.diagnostic.runtimePermissionStatus.DENIED:
                console.log("Permission denied - ask again");
                permisos();
                break;
            case cordova.plugins.diagnostic.runtimePermissionStatus.DENIED_ALWAYS:
                console.log("Permission permanently denied");
				myApp.alert('Salimos. Denegó permiso de localización', 'Localización', function () {
					navigator.app.exitApp();
				});
                break;
        }
    }, function(error){
        console.error("The following error occurred: "+error);
    }, cordova.plugins.diagnostic.runtimePermission.ACCESS_FINE_LOCATION);
}

function relocalizar() {
	console.log('Estoy dentro de RELOCALIZAR!!!');
	cordova.plugins.diagnostic.isLocationEnabled(function(enabled){
		if (estado == 0) {
			if (enabled) {
				//console.log('habilitado');
				// Estado 1 por si hay un resume que no vuelva a.
				estado = 1;
				lalocalizacion();
			} else {
				//console.log('deshabilitado');
				myApp.alert('Actívelo para situarle', 'GPS', function () {
					cordova.plugins.diagnostic.switchToLocationSettings();
				});
			}
		}
	}, function(error){
		console.error("GPS error: "+error);
	});
}

function lalocalizacion() {
	console.log('dentro de');
	var options = { maximumAge: 1, enableHighAccuracy: true };
	navigator.geolocation.getCurrentPosition(obtenida, noobtenida, options);
}

function obtenida(position) {
	lati = position.coords.latitude;
	longi = position.coords.longitude;
	losdatos.posicion = Base64.encode(lati +','+ longi);
}
```

LOGCAT
```
06-27 10:18:37.087 20420 20420 I chromium: [INFO:CONSOLE(37)] "ejecuto PERMISOS!!!", source: file:///android_asset/www/js/my-app.js (37)
06-27 10:18:37.114 20420 20420 I chromium: [INFO:CONSOLE(80)] "DB opened: olor.db", source: file:///android_asset/www/plugins/cordova-sqlite-storage/www/SQLitePlugin.js (80)
06-27 10:18:37.273 20420 20420 I chromium: [INFO:CONSOLE(216)] "CLOSE database: olor.db", source: file:///android_asset/www/plugins/cordova-sqlite-storage/www/SQLitePlugin.js (216)
06-27 10:18:37.273 20420 20420 I chromium: [INFO:CONSOLE(219)] "closing db with transaction queue length: 0", source: file:///android_asset/www/plugins/cordova-sqlite-storage/www/SQLitePlugin.js (219)
06-27 10:18:37.313 20420 20420 I chromium: [INFO:CONSOLE(136)] "cannot start next transaction: database not open", source: file:///android_asset/www/plugins/cordova-sqlite-storage/www/SQLitePlugin.js (136)
06-27 10:18:40.389 20420 20420 I chromium: [INFO:CONSOLE(44)] "Permission granted (or already granted) - call the plugin", source: file:///android_asset/www/js/my-app.js (44)
06-27 10:18:40.390 20420 20420 I chromium: [INFO:CONSOLE(46)] "Estoy dentro de GRANTED!!!!!", source: file:///android_asset/www/js/my-app.js (46)
06-27 10:18:40.405 20420 20420 I chromium: [INFO:CONSOLE(37)] "ejecuto PERMISOS!!!", source: file:///android_asset/www/js/my-app.js (37)
06-27 10:18:40.424 20420 20420 I chromium: [INFO:CONSOLE(44)] "Permission granted (or already granted) - call the plugin", source: file:///android_asset/www/js/my-app.js (44)
06-27 10:18:40.424 20420 20420 I chromium: [INFO:CONSOLE(46)] "Estoy dentro de GRANTED!!!!!", source: file:///android_asset/www/js/my-app.js (46)
06-27 10:18:40.466 20420 20420 I chromium: [INFO:CONSOLE(67)] "Estoy dentro de RELOCALIZAR!!!", source: file:///android_asset/www/js/my-app.js (67)
06-27 10:18:40.568 20420 20420 I chromium: [INFO:CONSOLE(67)] "Estoy dentro de RELOCALIZAR!!!", source: file:///android_asset/www/js/my-app.js (67)
```

**Comment by dpa99c** [comment]
When you call the plugin's `requestRuntimePermission()` function, it invokes the native Android runtime permissions API. This is asynchronous and invokes the permissions dialog activity, which puts your Cordova app's activity in the background (triggering a "pause" event). When the user responds to the dialog, the permissions activity is destroyed, putting your Cordova app activity back into the foreground (invoking the "resume" event). In parallel, the Android runtime permissions code invokes the native [`onRequestPermissionResult()`](https://github.com/dpa99c/cordova-diagnostic-plugin/blob/master/src/android/Diagnostic.java#L788) method with the result of the user interaction, and it is this that the plugin uses to invoke the Javascript callback with the result.

So in short, it's not possible to "combine" the resume event firing and the plugin result callback, because they are both a result of the behaviour of the native Android M runtime permissions code, which is part of the operating system.

What you can do is to write your app code in such a way as to expect and handle the concurrent invocation of the resume event and plugin success callback.

**Comment by gurumelo** [comment]
Thanks dpa99c
